Republik.ch.js

```diff
@@ -0,0 +1,388 @@
+{
+	"translatorID": "b4c83513-921f-474b-adc3-7017ea690ba7",
+	"label": "Republik.ch",
+	"creator": "Johannes Wüthrich",
+	"target": "^https?://www\\.republik\\.ch/[0-9]*/[0-9]*/[0-9]*/",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "g",
+	"lastUpdated": "2021-08-04 20:31:29"
+}
+
+/*
+Republik.ch Translator
+Copyright (C) 2021 Johannes Wüthrich
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+function detectWeb(doc, url) {
+	// TODO match different types of articles / debates / updates / blog-posts / newsletters etc.
+	
+	return "newspaperArticle";
+}
+
+
+function doWeb(doc, url) {
+	// TODO adapt for multiple type of sources (see above)
+	
+	scrape(doc, url);
+}
+
+
+function scrape(doc, url) {
+
+	var newItem = new Zotero.Item("newspaperArticle");
+
+	newItem.title = ZU.xpathText(doc, "//article//section[@class='title-block']/h1");
```

Comment:
If possible, use CSS selectors instead of XPath. This could be `text(doc, 'article section.title-block h1')` (and even `text(doc, 'article h1')` might suffice). More flexible because it doesn't match on the *entire* class attribute, just the presence of one particular class, and it's easier for people who haven't used XPath to understand.

Reply:
Changed in  46ae941